### ISSUE TITLE
lerna publish --independent triggering publish also for packages that haven't changed

### ISSUE THREAD

**Issue by marcofugaro** [lerna/lerna#1434]
When I run `lerna publish --independent`, lerna is asking me to publish also the packages that I haven't touched, and there is no way for me to skip the publishing of those.

From the readme: 
>  Each time you publish, you will get a prompt for *each package that has changed* to specify if it's a patch, minor, major or custom change.

## Expected Behavior
It should prompt me to publish only the packages that have changed.

## Current Behavior
It prompts the publish for every package.

## Steps to Reproduce (for bugs)
[This is the repo where this happens](https://github.com/accurat/accurapp/tree/big-updates)

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.1.2",
  "version": "independent",
  "npmClient": "yarn",
  "useWorkspaces": true
}
```
</p></details>

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.1.2 |
| `npm --version`  | 6.0.1 |
| `yarn --version` | 1.6.0 |
| `node --version` | 10.1.0 |

| OS | Version |
| --- | --- |
| macOS High Sierra | 10.13.4 |

**Comment by marcofugaro** [comment]
Nope, I didn't touch the dependencies either

**Comment by marcofugaro** [comment]
@evocateur right now only this

<img width="264" alt="screen shot 2018-05-24 at 11 06 57" src="https://user-images.githubusercontent.com/7217420/40475769-a7b441d6-5f42-11e8-9553-b5c1c3a7d906.png">

after I committed some changes to the package `accurapp-scripts`, the lerna diff is showing me the changed lines only of the files inside `accurapp-scripts` I changed, nothing else.

![image 1](https://user-images.githubusercontent.com/7217420/40479424-cfefdb66-5f4b-11e8-9266-6dbd28478410.png)


I tried to publish again after this, and it still prompts for every other package also.

**Comment by evocateur** [comment]
If every other local package depends on that package, they will also be bumped (because lerna bumps the dependency range of the package with changes). This is how lerna works.

**Comment by marcofugaro** [comment]
Except they don't, take for example the package `eslint-config-accurapp`, these are its [dependencies](https://github.com/accurat/accurapp/blob/big-updates/packages/eslint-config-accurapp/package.json):

<img width="568" alt="screen shot 2018-05-25 at 10 38 22" src="https://user-images.githubusercontent.com/7217420/40534904-c5fa912a-6007-11e8-8d11-90939529341f.png">

None of its dependencies is another package.

And this is lerna asking me to publish it, even if it isn't in the lerna diff:

<img width="666" alt="screen shot 2018-05-25 at 10 36 35" src="https://user-images.githubusercontent.com/7217420/40534935-e145de80-6007-11e8-815f-9406448a4142.png">

**Comment by evocateur** [comment]
All I can think of is that your tags are missing, somehow, and lerna thinks it has to publish everything. Might be because you last published on a branch that was then merged into master, which corrupts the tag references.

**Comment by marcofugaro** [comment]
Hmm interesting, might be, how can I try to fix it? By removing the tags?

**Comment by evocateur** [comment]
No, by ensuring that the tags exist, are reachable from your current HEAD, and such. `git describe` is an easy way to test “reachability.”

If the tags are missing or mangled, you’ll need to restore them (as annotated tags) on the master branch. Then lerna will properly ignore packages that have not changed since the last release (annotated git tag).

**Comment by marcofugaro** [comment]
@evocateur Thank you! I will check the tags.

Does lerna need EVERY past publish tags to be changed, or just the latest ones? Because the latest ones are already present in the same branch I'm publishing from (not master)

**Comment by evocateur** [comment]
As long as the most recent reachable tag is annotated (not lightweight), that’s all you need.

**Comment by marcofugaro** [comment]
Alright man, I checked the tags using [this script](https://stackoverflow.com/questions/40479712/how-can-i-tell-if-a-given-git-tag-is-annotated-or-lightweight/40480534#40480534) and turns out the latest tags are actually annotated. Also I made sure they're on the same branch I am in.

<img width="287" alt="screen shot 2018-05-30 at 12 02 56" src="https://user-images.githubusercontent.com/7217420/40713913-748ab038-6401-11e8-902c-1b43557d2e32.png">

I went ahead and made a diff for the latest tag and the current HEAD, only for one  package, like this:

```
git diff eslint-config-accurapp@4.1.0-beta.2..HEAD packages/eslint-config-accurapp/
```

And it outputs nothing because nothing has actually changed.

I don't know what else to do man, I provided you with the repo where this happens and the specific branch, if you wish to debug. Thanks for your help!
